### PR TITLE
add memory cached sources to portal profile

### PR DESCRIPTION
**Pull request by julsemaan**
# Description

Optimize fetching of portal profile sources
Adds a new in-memory cache to pfconfig that can be tied to a namespace
# Impacts

Portal profile sources
# Delete branch after merge

YES

### CODE REVIEW

**Comment by julsemaan** [comment]
For quick review and merge please, this is for v6

File: lib/pfconfig/memory_cached.pm [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+sub compute {
+    my ( $self, $key, $sub ) = @_;
+    my $logger = $self->logger;
+
+    my $result = $self->compute_from_subcache($key, sub {
+        return $sub->();
+    });
+
+    return $result;
+}
```
Comment (jrouzierinverse) [inline lib/pfconfig/memory_cached.pm]: Why not just call compute_from_subcache directly in the code as this is the same thing as
$self-> compute_from_subcache($key,  $sub);

Comment (julsemaan) [inline lib/pfconfig/memory_cached.pm]: Just thought it would be cleaner this way if we add additional logic to the computing.

Comment (rouzier) [inline lib/pfconfig/memory_cached.pm]: This will be called often IMO it should be kept "light" until we need that extra layer of abstraction.

**Comment by julsemaan** [comment]
re-review

**Pull request merged**